python/tank_vendor/shotgun_deploy/descriptor_io/git.py

```diff
@@ -0,0 +1,337 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+import os
+import copy
+import uuid
+import tempfile
+
+from ..util import subprocess_check_output, execute_git_command
+from .base import IODescriptorBase
+from ..zipfilehelper import unzip_file
+from ..errors import ShotgunDeployError
+from ...shotgun_base import ensure_folder_exists, safe_delete_file
+
+from .. import util
+
+log = util.get_shotgun_deploy_logger()
+
+class IODescriptorGit(IODescriptorBase):
+    """
+    Represents a tag in a git repository.
+
+    location: {"type": "git", "path": "/path/to/repo.git", "version": "v0.2.1"}
+
+    path can be on the form:
+
+        git@github.com:manneohrstrom/tk-hiero-publish.git
+        https://github.com/manneohrstrom/tk-hiero-publish.git
+        git://github.com/manneohrstrom/tk-hiero-publish.git
+        /full/path/to/local/repo.git
+
+    Uris are on the form:
+
+        sgtk:git:path/to/git/repo:v12.3.4
```

Comment:
should this be `sgtk:git:/path/to/git/repo:v12.3.4`? Note the `/` before `path`.